crates/sui-core/src/consensus_adapter.rs

```diff
@@ -452,20 +597,78 @@ impl ConsensusAdapter {
             metrics.sequencing_certificate_success.inc();
         });
     }
+
+    async fn submit_after_selected(
+        self: &Arc<Self>,
+        transaction: ConsensusTransaction,
+        epoch_store: &Arc<AuthorityPerEpochStore>,
+    ) {
+        let transaction_key = transaction.key();
+        debug!("Submitting {transaction_key:?} to consensus");
+
+        {
+            let ack_start = Instant::now();
+            let mut retries: u32 = 0;
+            while let Err(e) = self
+                .consensus_client
+                .submit_to_consensus(&transaction, epoch_store)
+                .await
+            {
+                // This can happen during Narwhal reconfig, so wait for a few retries.
+                if retries > 3 {
+                    error!(
+                        "Error submitting transaction to own narwhal worker: {:?}",
+                        e
+                    );
+                }
+                if let Some(metrics) = self.opt_metrics.as_ref() {
+                    metrics.sequencing_certificate_failures.inc();
+                }
+                retries += 1;
+                time::sleep(Duration::from_secs(10)).await;
```

Comment:
As an extra optimisation - which we didn't have before (so we might do it on follow up PR) - we can probably pass in the `process_waiter` and do a `select` between the `process_waiter` and the `timeout`. If the `process_waiter` has returned successfully, then there is no need for us to keep waiting and trying to submit in consensus. Or alternatively we can even do this in a higher level (when we execute the `submit_after_selected` method)